agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4415,6 +4415,20 @@ bool slave_thread::autoconfig_wsc_parse_m2_encrypted_settings(
 
     // Swap back to host byte order to read and use config_data
     config_data.class_swap();
+
+    // TODO tlvf should check this
+    if (config_data.ssid_length() > WSC::WSC_MAX_SSID_LENGTH) {
+        LOG(INFO) << "SSID too long: " << config_data.ssid_length();
+        return false;
+    }
+    // TODO tlvf should do conversion to std::string
```

Comment:
See #265 - you can take over if needed or we can merge it and you can rebase.

Reply:
I'll see after #265 has been merged.